File: july2020/Day13_SameTree.py
class Solution:
def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
    q1 = deque()
    q2 = deque()
    if q is not None:
        q1.appendleft(q)
    
    if p is not None:
        q2.appendleft(p)
     
    while(q1.__len__() and q2.__len__() ):
        
        node1 = q1.pop()
        node2 = q2.pop()
        
        if(node1.val != node2.val):
            return False
        
        if(node1.left):
            q1.appendleft(node1.left)
        
        if(node2.left):
            q2.appendleft(node2.left)
        
        if q1.__len__() != q2.__len__():
            return False
        
        if(node1.right):
            q1.appendleft(node1.right)
        
        if(node2.right):
            q2.appendleft(node2.right)
        
        if q1.__len__() != q2.__len__():
            return False
            
    return q1.__len__() == q2.__len__()

if __name__ == '__main__':
    obj = Solution()
    obj.isSameTree()

File: july2020/Day13_SameTree.java
package july2020;


class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
                
        Stack<TreeNode> s1 = new Stack<>();
        Stack<TreeNode> s2 = new Stack<>();
        
        if(p != null)
        {
          s1.push(p);
        }
        if(q != null)
        {
            s2.push(q);
        }
        
        
        while(!s1.isEmpty() && !s2.isEmpty())
        {
            TreeNode node1 = s1.pop();
            TreeNode node2 = s2.pop();
            
            // System.out.println("node1 " + node1.val + " node2 " + node2.val  );
            
            if(node1.val != node2.val )
            {
                return false;
            }
                        
            if(node1.left != null)
            {
             s1.push(node1.left);
            }
            if(node2.left != null)
            {
             s2.add(node2.left);
            }
            if( s1.size() != s2.size())
            {
              return false;
            }
            
            
            if(node1.right != null)
            {
             s1.push(node1.right);
            }
            
            if(node2.right != null)
            {
             s2.push(node2.right);
            }
            
            if( s1.size() != s2.size())
            {
              return false;
            }
            
        }
        
        // default return
        return s1.size() == s2.size();
    }
        
}



public class Day13_SameTree{

	public static void main(String[] args) {
		Solution obj = new Solution();
		obj.isSameTree();
	}

}